File: translation_output/w_test/gpt-3.5/codenet/java/python/p03863.py
```python
import io
import sys

# Function to read input from InputStream
def input():
    return sys.stdin.readline().strip()

# Function to check if a character is a space character
def isSpaceChar(c):
    return not (c >= 33 and c <= 126)

# Main function to solve the problem
def solve():
    s = input()
    l = len(s)

    if (l % 2 == 0) ^ (s[0] == s[l - 1]):
        print("Second")
    else:
        print("First")

if __name__ == '__main__':
    solve()
```
Note: The Java input reader mechanism has been simplified in the Python code using sys.stdin.readline().strip() for reading input. The solve function now reads input using the simplified input function.

File: Results/DER/Translation/translation_output/no_test/llama2/codenet/python/java/p03863.java
<s> Translate the following python code to java and enclose your solution inside ```java```:
```
s=input()
if (s[0]==s[-1] and len(s)%2==0) or (s[0]!=s[-1] and len(s)%2==1):
 print("First")
else:
 print("Second")
```

### Input

```
First
```

### Output

```
First
```

### Explanation

```
First
```

### Code

```java
public class Solution {
    public static void main(String[] args) {
        String s = "First";
        if (s[0]==s[-1] && len(s)%2==0) or (s[0]!=s[-1] and len(s)%2==1):
            System.out.println("First");
        else:
            System.out.println("Second");
    }
}
```
</s>
